pass-data-client/src/test/java/org/eclipse/pass/support/client/SubmissionStatusCalculatorTest.java

```diff
@@ -0,0 +1,571 @@
+        assertEquals(SubmissionStatus.MANUSCRIPT_REQUIRED,
+                SubmissionStatusCalculator.calculatePreSubmissionStatus(null, null));
+        assertEquals(SubmissionStatus.MANUSCRIPT_REQUIRED,
+                     SubmissionStatusCalculator.calculatePreSubmissionStatus(new ArrayList<SubmissionEvent>(), null));
+        assertEquals(SubmissionStatus.MANUSCRIPT_REQUIRED,
+                     SubmissionStatusCalculator.calculatePreSubmissionStatus(null,
+                             SubmissionStatus.MANUSCRIPT_REQUIRED));
+        assertEquals(SubmissionStatus.MANUSCRIPT_REQUIRED,
+                     SubmissionStatusCalculator.calculatePreSubmissionStatus(Collections.emptyList(),
+                             SubmissionStatus.MANUSCRIPT_REQUIRED));
+    }
+
+    @Test
+    public void testPreSubmissionStatusDraftExpected() {
+        assertEquals(SubmissionStatus.DRAFT, SubmissionStatusCalculator.calculatePreSubmissionStatus(null,
+                SubmissionStatus.DRAFT));
+
+        // in fact, the logic inside the calculator will default to supplied status if there are no submission events
+        // provided, rather than hard-coding a return.
+        assertEquals(SubmissionStatus.SUBMITTED, SubmissionStatusCalculator.calculatePreSubmissionStatus(null,
+                SubmissionStatus.SUBMITTED));
+    }
+
+    /**
+     * Tests various scenarios where outcome should be APPROVAL_REQUESTED
+     */
+    @Test
+    public void testPreSubmissionStatusApprovalRequested() {
+        List<SubmissionEvent> submissionEvents =
+            Arrays.asList(submissionEvent(ZonedDateTime.now(), EventType.APPROVAL_REQUESTED));
+        assertEquals(SubmissionStatus.APPROVAL_REQUESTED,
+                     SubmissionStatusCalculator.calculatePreSubmissionStatus(submissionEvents, null));
+
+        submissionEvents =
+            Arrays.asList(submissionEvent(ZonedDateTime.now(), EventType.APPROVAL_REQUESTED_NEWUSER));
+        assertEquals(SubmissionStatus.APPROVAL_REQUESTED,
+                     SubmissionStatusCalculator.calculatePreSubmissionStatus(submissionEvents, null));
+
+        submissionEvents =
+            Arrays.asList(submissionEvent(ZonedDateTime.now(), EventType.APPROVAL_REQUESTED_NEWUSER),
+                          submissionEvent(ZonedDateTime.now(), EventType.CHANGES_REQUESTED),
+                          submissionEvent(ZonedDateTime.now(), EventType.APPROVAL_REQUESTED));
+        assertEquals(SubmissionStatus.APPROVAL_REQUESTED,
+                     SubmissionStatusCalculator.calculatePreSubmissionStatus(submissionEvents, null));
+
+        assertEquals(SubmissionStatus.APPROVAL_REQUESTED,
+                     SubmissionStatusCalculator.calculatePreSubmissionStatus(submissionEvents, SubmissionStatus.DRAFT));
+    }
+
+    /**
+     * Tests various scenarios in which the pre-submission status should be "CHANGES_REQUESTED"
+     */
+    @Test
+    public void testPreSubmissionStatusChangesRequested() {
```

Comment:
When doing a mvn clean install I get this test failure:

[ERROR] testPreSubmissionStatusChangesRequested  Time elapsed: 0 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <CHANGES_REQUESTED> but was: <APPROVAL_REQUESTED>
        at org.eclipse.pass.support.client.SubmissionStatusCalculatorTest.testPreSubmissionStatusChangesRequested(SubmissionStatusCalculatorTest.java:441)

Reply:
Interesting. A failure in a unit test is strange. I just tried again locally and it passes. It's also passing in github. I wonder if there is somehow a platform dependency. Could you do a git pull on the branch just to make sure you have everything?

Reply:
Ah. I bet the problem is the use of ZonedDateTime.now() in the unit tests. The service is sorting based on most recent time.

Reply:
I doubled checked, 561-sub-status is up to date with origin. Ran git oull for funsies, and it is already up to date.

This is the full error message:

[INFO] Running org.eclipse.pass.support.client.SubmissionStatusCalculatorTest
10:41:58.283 [main] WARN org.eclipse.pass.support.client.SubmissionStatusCalculator - The current status of the Submission conflicts with the status calculated based on the most recent SubmissionEvent. The status on the Submission record is `APPROVAL_REQUESTED`, while the calculated status is `CHANGES_REQUESTED`. The UI is responsible
 for setting pre-Submission statuses, but this mismatch may indicate a data issue.
10:41:58.315 [main] WARN org.eclipse.pass.support.client.SubmissionStatusCalculator - The current status of the Submission conflicts with the status calculated based on the most recent SubmissionEvent. The status on the Submission record is `APPROVAL_REQUESTED`, while the calculated status is `CHANGES_REQUESTED`. The UI is responsible
 for setting pre-Submission statuses, but this mismatch may indicate a data issue.
10:41:58.315 [main] WARN org.eclipse.pass.support.client.SubmissionStatusCalculator - The current status of the Submission conflicts with the status calculated based on the most recent SubmissionEvent. The status on the Submission record is `CHANGES_REQUESTED`, while the calculated status is `APPROVAL_REQUESTED`. The UI is responsible
 for setting pre-Submission statuses, but this mismatch may indicate a data issue.
10:41:58.315 [main] WARN org.eclipse.pass.support.client.SubmissionStatusCalculator - The current status of the Submission conflicts with the status calculated based on the most recent SubmissionEvent. The status on the Submission record is `CHANGES_REQUESTED`, while the calculated status is `CANCELLED`. The UI is responsible for sett
ing pre-Submission statuses, but this mismatch may indicate a data issue.
10:41:58.315 [main] WARN org.eclipse.pass.support.client.SubmissionStatusCalculator - The current status of the Submission conflicts with the status calculated based on the most recent SubmissionEvent. The status on the Submission record is `DRAFT`, while the calculated status is `APPROVAL_REQUESTED`. The UI is responsible for setting
 pre-Submission statuses, but this mismatch may indicate a data issue.
10:41:58.315 [main] WARN org.eclipse.pass.support.client.SubmissionStatusCalculator - The current status of the Submission conflicts with the status calculated based on the most recent SubmissionEvent. The status on the Submission record is `DRAFT`, while the calculated status is `CHANGES_REQUESTED`. The UI is responsible for setting 
pre-Submission statuses, but this mismatch may indicate a data issue.
10:41:58.315 [main] WARN org.eclipse.pass.support.client.SubmissionStatusCalculator - The current status of the Submission conflicts with the status calculated based on the most recent SubmissionEvent. The status on the Submission record is `DRAFT`, while the calculated status is `CANCELLED`. The UI is responsible for setting pre-Subm
ission statuses, but this mismatch may indicate a data issue.
[ERROR] Tests run: 21, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.119 s <<< FAILURE! - in org.eclipse.pass.support.client.SubmissionStatusCalculatorTest
[ERROR] testPreSubmissionStatusCancelled  Time elapsed: 0.001 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <CANCELLED> but was: <APPROVAL_REQUESTED>
        at org.eclipse.pass.support.client.SubmissionStatusCalculatorTest.testPreSubmissionStatusCancelled(SubmissionStatusCalculatorTest.java:469)

[ERROR] testPreSubmissionStatusChangesRequested  Time elapsed: 0 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <CHANGES_REQUESTED> but was: <APPROVAL_REQUESTED>
        at org.eclipse.pass.support.client.SubmissionStatusCalculatorTest.testPreSubmissionStatusChangesRequested(SubmissionStatusCalculatorTest.java:441)

[INFO] Running org.eclipse.pass.support.client.SubmissionStatusServiceTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.261 s - in org.eclipse.pass.support.client.SubmissionStatusServiceTest
[INFO] 
[INFO] Results:
[INFO]
[ERROR] Failures: 
[ERROR]   SubmissionStatusCalculatorTest.testPreSubmissionStatusCancelled:469 expected: <CANCELLED> but was: <APPROVAL_REQUESTED>
[ERROR]   SubmissionStatusCalculatorTest.testPreSubmissionStatusChangesRequested:441 expected: <CHANGES_REQUESTED> but was: <APPROVAL_REQUESTED>